### PR TITLE
🎉Source Stripe: New Stream - Subscription Schedule

### LINKED ISSUE #16980: Source Stripe: new stream Subscription Schedule
## Tell us about the problem you're trying to solve
[Subscription Schedule](https://stripe.com/docs/api/subscription_schedules)
A subscription schedule allows you to create and manage the lifecycle of a subscription by predefining expected changes. 

[Permission](https://stripe.com/docs/security/permissions)

## Describe the solution you’d like
Research pros and cons for implementing this stream.
_Research should contain: what stream should be implemented, permissions and scopes (if needed), paid plan (if needed)._

### PR DESCRIPTION
**Pull request by btkcodedev**
## What
Closes #16980

## Enhancement
New streams: 

- Subscription Schedule
- Subscription Schedule with ID

## Recommended reading order
1. `source_stripe/source.py`
2. `source_stripe/streams.py`
3.  `source_stripe/schemas/[subscription_schedule]`

## 🚨 User Impact 🚨
No breaking changes

## Pre-merge Checklist
<details open><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter [Yes]

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Issue acceptance criteria met
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [x] Documentation which references the generator is updated as needed

## For the reviewer: 
Enhancement passing in local python integration tests. Needs only to test deployment to cloud build and publish it :)

### CODE REVIEW

**Comment by YowanR** [comment]
Can you please review this @sajarin @marcosmarxm @sh4sh ? Thanks 🙏

**Comment by btkcodedev** [comment]
@marcosmarxm @sajarin @sh4sh Little bump here :)

**Comment by btkcodedev** [comment]
@sajarin oops what went wrong? Your thoughts please🙂

File: airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml [hunk 1]
```diff
@@ -29,6 +29,10 @@ acceptance_tests:
             bypass_reason: "This stream can't be seeded in our sandbox account"
           - name: "checkout_sessions_line_items"
             bypass_reason: "This stream can't be seeded in our sandbox account"
+          - name: "subscription_schedule"
+            bypass_reason: "This stream can't be seeded in our sandbox account"
+          - name: "subscription_schedule_with_id"
+            bypass_reason: "This stream can't be seeded in our sandbox account"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml:35]: You're adding new streams to a GA connector. We **must** test them with data.

Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml:35]: Remove this.

File: airbyte-integrations/connectors/source-stripe/source_stripe/schemas/subscription_schedule.json [hunk 2]
```diff
@@ -0,0 +1,240 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Generated schema for Root",
+    "type": "object",
+    "properties": {
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-stripe/source_stripe/schemas/subscription_schedule.json:5]: The schema is not correct, @btkcodedev please check others connector and correct it.

File: airbyte-integrations/connectors/source-stripe/source_stripe/schemas/subscription_schedule_with_id.json [hunk 3]
```diff
@@ -0,0 +1,214 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Generated schema for Root",
+    "type": "object",
+    "properties": {
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-stripe/source_stripe/schemas/subscription_schedule_with_id.json:5]: Same here.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Left some comments.

**Comment by marcosmarxm** [comment]
@btkcodedev please check failures.

**Comment by btkcodedev** [comment]
I've added the empty stream, but my local still shows that at least one record should be read with DockerContainer Error.
Is the change okay? Please let me know

**Comment by sajarin** [comment]
Hey thanks for the contribution and apologies for the delay! 

I'll be the DRI responsible for reviewing this and getting it across the finish line. We currently have a backlog of about a dozen PRs, but just wanted to let you know that this PR has been added to the queue and I'll be trying my best to get to this as soon as possible. 

Thanks for the contribution and for being patient :)

**Comment by btkcodedev** [comment]
Thanks, It is ready for review,
Please forward

File: airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml [hunk 4]
```diff
@@ -71,3 +71,4 @@ acceptance_tests:
               bypass_reason: "URL changes upon each request for privacy/security"
             - name: hosted_invoice_url
               bypass_reason: "URL changes upon each request for privacy/security"
+
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml:74]: EOF

File: airbyte-integrations/connectors/source-stripe/source_stripe/schemas/subscription_schedule.json [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    "current_phase": {
+      "type": ["null", "string"]
+    },
+    "customer": {
+      "type": ["null", "string"]
+    },
+    "default_settings": {
+      "type": ["null", "object"],
+      "properties": {
+        "application_fee_percent": {
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-stripe/source_stripe/schemas/subscription_schedule.json:32]:
```suggestion
    "default_settings": {
      "type": ["null", "object"],
      "additionalProperties": true,
      "properties": {
        "application_fee_percent": {
```

**Review by marcosmarxm — APPROVED** [review]
Thanks @btkcodedev the connector will publish during the week.

File: docs/integrations/sources/jira.md [hunk 6]
```diff
@@ -126,7 +126,7 @@ The Jira connector should not run into Jira API limitations under normal usage.
 
 | Version | Date       | Pull Request                                                | Subject                                                                                                                 |
 |:--------|:-----------|:------------------------------------------------------------|:------------------------------------------------------------------------------------------------------------------------|
-| 0.3.6   | 2023-04-10 | [\#24636](https://github.com/airbytehq/airbyte/pull/24636)  | Removed Connector Domain Pattern from Spec                                                                              |
+| 0.3.6   | 2023-04-10 | [\#24636](https://github.com/airbytehq/airbyte/pull/24636)  | Remove pattern for domain check in the specification                                                                          |
```
Comment (bazarnov) [inline docs/integrations/sources/jira.md:129]: Why do we have this filed modified? It's related to the other connector (source-jira).

File: docs/integrations/sources/stripe.md [hunk 7]
```diff
@@ -82,8 +83,7 @@ The Stripe connector should not run into Stripe API limitations under normal usa
 
 | Version | Date       | Pull Request                                             | Subject                                                                                                                                              |
 |:--------|:-----------|:---------------------------------------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------|
-| 3.0.5   | 2023-03-25 | [22866](https://github.com/airbytehq/airbyte/pull/22866) | Specified date formatting in specification                                                     |
-| 3.0.4   | 2023-03-24 | [24471](https://github.com/airbytehq/airbyte/pull/24471) | Fix stream slices for single sliced streams                                                                                                          |
+| 3.1.0   | 2023-04-10 | [23624](https://github.com/airbytehq/airbyte/pull/23624) | Add new stream `Subscription Schedule`                                                                                                               |
```
Comment (bazarnov) [inline docs/integrations/sources/stripe.md:86]: If we add the new stream this is not a breaking change, please make version `3.0.4` a minor, non-breaking-change update (patch basically).

File: airbyte-config-oss/init-oss/src/main/resources/seed/source_definitions.yaml [hunk 8]
```diff
@@ -2006,7 +2006,7 @@
 - name: Stripe
   sourceDefinitionId: e094cb9a-26de-4645-8761-65c0c425d1de
   dockerRepository: airbyte/source-stripe
-  dockerImageTag: 3.0.5
+  dockerImageTag: 3.1.0
```
Comment (bazarnov) [inline airbyte-config-oss/init-oss/src/main/resources/seed/source_definitions.yaml:2009]: 3.0.4

**Review by bazarnov — CHANGES_REQUESTED** [review]

File: airbyte-config-oss/init-oss/src/main/resources/seed/source_definitions.yaml (hunk 8 above)
Comment (marcosmarxm) [reply airbyte-config-oss/init-oss/src/main/resources/seed/source_definitions.yaml:2009]: @bazarnov we're following the guidelines in contributing to airbyte made by the connector team https://docs.airbyte.com/contributing-to-airbyte/#examples

**Review by bazarnov — APPROVED** [review]

**Pull request merged**